apps/ffcniftya/app.js

```diff
@@ -30,77 +11,50 @@ function ISO8601_week_no(date) {
   return 1 + Math.ceil((firstThursday - tdt) / 604800000);
 }
 
-function d02(value) {
-  return ('0' + value).substr(-2);
+function format(value) {
+  return ("0" + value).substr(-2);
 }
 
-function draw() {
-  g.reset();
-  g.clearRect(0, widget, viewport.width, viewport.height);
-  const now = new Date();
+const ClockFace = require("ClockFace");
+const clock = new ClockFace({
+  init: function () {
+    const appRect = Bangle.appRect;
 
-  const hour = d02(now.getHours() - (is12Hour && now.getHours() > 12 ? 12 : 0));
-  const minutes = d02(now.getMinutes());
-  const day = d02(now.getDate());
-  const month = d02(now.getMonth() + 1);
-  const year = now.getFullYear(now);
-  const weekNum = d02(ISO8601_week_no(now));
-  const monthName = locale.month(now, 3);
-  const dayName = locale.dow(now, 3);
+    this.viewport = {
+      width: appRect.w,
+      height: appRect.h
+    };
 
-  const centerTimeScaleX = center.x + 32 * scale;
-  g.setFontAlign(1, 0).setFont("Vector", 90 * scale);
-  g.drawString(hour, centerTimeScaleX, center.y - 31 * scale);
-  g.drawString(minutes, centerTimeScaleX, center.y + 46 * scale);
+    this.center = {
+      x: this.viewport.width / 2,
+      y: Math.round((this.viewport.height / 2) + appRect.y)
+    };
 
-  g.fillRect(center.x + 30 * scale, center.y - 72 * scale, center.x + 32 * scale, center.y + 74 * scale);
+    this.scale = g.getWidth() / this.viewport.width;
+    this.centerTimeScaleX = this.center.x + 32 * this.scale;
+    this.centerDatesScaleX = this.center.x + 40 * this.scale;
 
-  const centerDatesScaleX = center.x + 40 * scale;
-  g.setFontAlign(-1, 0).setFont("Vector", 16 * scale);
-  g.drawString(year, centerDatesScaleX, center.y - 62 * scale);
-  g.drawString(month, centerDatesScaleX, center.y - 44 * scale);
-  g.drawString(day, centerDatesScaleX, center.y - 26 * scale);
-  if (showWeekNum) g.drawString(weekNum, centerDatesScaleX, center.y + 15 * scale);
-  g.drawString(monthName, centerDatesScaleX, center.y + 48 * scale);
-  g.drawString(dayName, centerDatesScaleX, center.y + 66 * scale);
-}
+    this.showWeekNum = Object.assign({ showWeekNum: true }, require("Storage").readJSON("ffcniftya.json", true))["showWeekNum"];
```

Comment:
I don't think this does what you want it to do: it always sets `this.showWeekNum` to an object, so `if (this.showWeekNum)` below will always evaluate to true.

What you could do is
* specify `settingsFile: "ffcniftya.json",` to have ClockFace read in the settings for you
* check for `if (this.showWeekNum !== false)` below, (making it default to enabled if not set).

Reply:
> I don't think this does what you want it to do: it always sets this.showWeekNum to an object, so if (this.showWeekNum) below will always evaluate to true.

I just tried to change the setting and it works. Maybe you didn't notice the ..["showWeekNum"] at the end?

Reply:
Whoops, you're right, sorry.

Reply:
No problem! 😄